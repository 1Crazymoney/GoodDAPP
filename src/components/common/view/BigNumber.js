// @flow
import React from 'react'
import { View } from 'react-native'
import normalize from 'react-native-elements/src/helpers/normalizeText'
import { withStyles } from '../../../lib/styles'
import Text from './Text'

/**
 * Receives a number and a unit to display
 * @param {Props} props
 * @param {Number} [props.number] Number to show
 * @param {String} [props.unit] Units for the number
 * @param {Object} [props.elementStyles] Inner elements styles
 * @param {Object} [props.style] Outer element style
 * @returns {React.Node}
 */
class BigNumber extends React.Component {
  render() {
    const { elementStyles, number, unit, style, styles } = this.props

    return (
      <View style={[styles.bigNumberWrapper, style]}>
        <Text style={[styles.bigNumber, elementStyles]}>{number}</Text>
        <Text style={[styles.bigNumberUnit, elementStyles]}>{unit}</Text>
      </View>
    )
  }
}

const getStylesFromProps = ({ theme }) => {
  return {
    bigNumberWrapper: {
      display: 'inline-block'
    },
    bigNumber: {
      ...theme.fontStyle,
      fontSize: normalize(30),
      textAlign: 'right'
    },
    bigNumberUnit: {
      ...theme.fontStyle,
      textAlign: 'right'
    }
  }
}

export default withStyles(getStylesFromProps)(BigNumber)
