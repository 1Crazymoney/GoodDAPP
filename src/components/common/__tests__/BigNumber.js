import React from 'react'
import BigNumber from '../BigNumber'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('BigNumber', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<BigNumber number={10} unit="GD" />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<BigNumber number={10} unit="GD" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
