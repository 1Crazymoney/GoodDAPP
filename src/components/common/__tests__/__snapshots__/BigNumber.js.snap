// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigNumber matches snapshot 1`] = `
<div
  style={
    Object {
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexDirection": "column",
      "alignItems": "stretch",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "display": "inline-block",
      "flexDirection": "column",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    dir="auto"
    style={
      Object {
        "borderBottomWidth": "0px",
        "borderLeftWidth": "0px",
        "borderRightWidth": "0px",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "color": "rgba(85,85,85,1.00)",
        "direction": "ltr",
        "display": "inline",
        "fontFamily": "Helvetica, \\"sans-serif\\"",
        "fontSize": "30px",
        "fontStyle": "inherit",
        "fontVariant": "inherit",
        "fontWeight": "inherit",
        "lineHeight": "inherit",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "textAlign": "right",
        "textDecoration": "none",
        "whiteSpace": "pre-wrap",
        "wordWrap": "break-word",
      }
    }
  >
    10
  </div>
  <div
    dir="auto"
    style={
      Object {
        "borderBottomWidth": "0px",
        "borderLeftWidth": "0px",
        "borderRightWidth": "0px",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "color": "rgba(85,85,85,1.00)",
        "direction": "ltr",
        "display": "inline",
        "fontFamily": "Helvetica, \\"sans-serif\\"",
        "fontSize": "18px",
        "fontStyle": "inherit",
        "fontVariant": "inherit",
        "fontWeight": "inherit",
        "lineHeight": "inherit",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "textAlign": "right",
        "textDecoration": "none",
        "whiteSpace": "pre-wrap",
        "wordWrap": "break-word",
      }
    }
  >
    GD
  </div>
</div>
`;
